packages/quantic/force-app/main/default/lwc/quanticRecommendationList/quanticRecommendationList.js

```diff
@@ -0,0 +1,199 @@
+   * @default {3}
+   */
+  @api numberOfRecommendationsPerRow = 3;
+  /**
+   * A list of fields to include in the query results, separated by commas.
+   * @api
+   * @type {string}
+   * @defaultValue `'date,author,source,language,filetype,parents,sfknowledgearticleid,sfid,sfkbid,sfkavid'`
+   */
+  @api fieldsToInclude =
+    'date,author,source,language,filetype,parents,sfknowledgearticleid,sfid,sfkbid,sfkavid';
+  /**
+   * The label of the component. This label is displayed in the component header.
+   * @api
+   * @type {string}
+   */
+  @api label;
+  /**
+   * The Heading level to use for the heading label, accepted values are integers from 1 to 6.
+   * @type {number}
+   * @default {1}
+   */
+  @api headingLevel = 1;
+
+  /** @type {RecommendationListState} */
+  @track state;
+  /** @type {RecommendationList} */
+  recommendationList;
+  /** @type {boolean} */
+  showPlaceholder = false;
+  /** @type {Function} */
+  unsubscribe;
+  /** @type {ResultTemplatesManager} */
+  resultTemplatesManager;
+  /** @type {CoveoHeadlessRecommendation} */
+  headless;
+  /** @type {boolean} */
+  hasInitializationError = false;
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+    this.setRecommendationWidth();
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  /**
+   * @param {RecommendationEngine} engine
+   */
+  initialize = (engine) => {
+    this.headless = getHeadlessBundle(this.engineId);
+    this.recommendationList = this.headless.buildRecommendationList(engine, {
+      options: {
+        id: this.recommendation,
+        numberOfRecommendations: Number(this.numberOfRecommendations),
+      },
+    });
+    this.unsubscribe = this.recommendationList.subscribe(() =>
+      this.updateState()
+    );
+
+    this.resultTemplatesManager =
+      this.headless.buildResultTemplatesManager(engine);
+    this.registerTemplates();
+
+    this.actions = {
+      ...this.headless.loadFieldActions(engine),
+    };
+
+    engine.dispatch(this.actions.registerFieldsToInclude(this.fields));
+  };
+
+  registerTemplates() {
+    this.resultTemplatesManager.registerTemplates({
+      content: defaultRecommendationTemplate,
+      conditions: [],
+    });
+    this.dispatchEvent(
+      new CustomEvent('registerrecommendationtemplates', {
+        bubbles: true,
+        detail: this.resultTemplatesManager,
+      })
+    );
+  }
+
+  disconnectedCallback() {
+    this.unsubscribe?.();
+  }
+
+  updateState() {
+    this.state = this.recommendationList?.state;
+    this.showPlaceholder =
+      this.recommendationList?.state?.isLoading &&
+      !this.recommendationList?.state?.recommendations?.length &&
+      !this.recommendationList?.state?.error;
+  }
+
+  setRecommendationWidth() {
+    const styles = this.template.host?.style;
+    styles.setProperty(
+      '--recommendationItemWidth',
+      `${100 / this.numberOfRecommendationsPerRow}%`
+    );
+  }
```

Comment:
This could work yes however it will introduce some sort of incoherence, in my opinion,  in the case where the end user wants to display more than 12 units per row. I know it's kinda unlikely to happen but in the case  where for example the property  `recommendations-per-row` is set to a value larger than `12` the end user would expect more than 12 items displayed per row but instead he will get only 12 per row which will lead to this incoherence between the property passed and the result displayed.

This also applies to the case where we display a carousel, if the end user wants to display  for example `13 `recommendations per page, there is no class such as `slds-size_1-of-13`.

Additionally, my method is following the same logic as these SLDS Col Classes  cause there is nothing really magic about for example `slds-size_6-of-12`, the only thing it does is setting the width to `50%` which is exactly what I'm doing. But I'm also removing the incoherence I mentioned above.

If the end user wants to display the recommendations in a particular way that would allow him to display more than 12 recommendations per row, he can.

Reply:
Okay to reproduce the logic then 👍 especially if it looks like the same logic slds is using.

I do think it would be fair to simply document that in a "non-carousel" variant, the limit would be 12 per row and I don't think it would cause issues.

But we will leave it like you've done it so far, sounds good.